lib/utils/dialect.ts

```diff
@@ -0,0 +1,69 @@
+import isPlainObject from 'lodash/isPlainObject';
+// eslint-disable-next-line import/order
+import { v1 as uuidv1, v4 as uuidv4 } from 'uuid';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const DataTypes = require('../data-types');
+
+const dialects = new Set([
```

Comment:
Snowflake not needed for this one?

Reply:
Looks like an oversight from the Snowflake PR, because their DataTypes support milliseconds.

Although really this should be a flag in Dialect.supports.

I'll note it down and do a follow-up PR with it just in case there is something fishy going on.